pgcli/pgcompleter.py

```diff
@@ -275,6 +275,8 @@ def extend_foreignkeys(self, fk_data):
             parentschema, childschema = e([fk.parentschema, fk.childschema])
             parenttable, childtable = e([fk.parenttable, fk.childtable])
             childcol, parcol = e([fk.childcolumn, fk.parentcolumn])
+            if childtable not in meta[childschema]:
```

Comment:
Hmm. This will fix the exception, but I wonder why the key would not be present? Is it possible there's an issue when building a dict of `meta[childschema]`? If so, we better fix the root cause.

Reply:
The root issue comes from this place (only when you have partitioned tables)
https://github.com/dbcli/pgcli/blob/master/pgcli/pgexecute.py#L521-L524


For partitioned tables `relkind` equals `p`, so for partitioned tables key is not there because we do not put it there.

Starting from 9.6+ `relkind` has new members (more info here: https://www.postgresql.org/docs/12/catalog-pg-class.html) 

relkind | char |   | r = ordinary table, i = index, S = sequence, t = TOAST table, v = view, m = materialized view, c = composite type, f = foreign table, p = partitioned table, I = partitioned index
-- | -- | -- | --

I've also tried to include new relkind to query metadata and it worked. But I'm not sure if we should treat partitioned table as a table or not.

Reply:
I think we should include new table types.

Reply:
I'm not sure about (t) and (f) types but I've added changes to introduce (p) type. I've been running pgcli with these modifications for a week and it looks good so far.

Reply:
@troyanov Can you take a look at https://github.com/dbcli/pgcli/pull/1145? The PR author thinks we should probably add `f` but not `t`, and it sounds sensible to me as well.

Reply:
@j-bennet I've never worked with `f` and `t` table types, not sure if I have correct knowledge. But According to the docs I've read PR author is correct about `t` type that it should not be added.